app/src/main/java/pm/gnosis/heimdall/ui/safe/overview/SafeAdapter.kt

```diff
@@ -1,49 +1,145 @@
+    }
+
+    inner class ViewHolder(itemView: View) : CastingViewHolder<Safe>(Safe::class.java, itemView), View.OnClickListener {
+
         init {
             itemView.setOnClickListener(this)
             itemView.layout_safe_item_share.setOnClickListener {
-                items[adapterPosition].address.asEthereumAddressString().let {
-                    val title = context.getString(R.string.sharing_x, items[adapterPosition].name ?: "")
-                    context.shareExternalText(it, title)
+                currentEntry?.let {
+                    it.let {
+                        val addressString = it.address.asEthereumAddressString()
+                        val title = context.getString(R.string.sharing_x, it.name ?: addressString)
+                        context.shareExternalText(addressString, title)
+                    }
                 }
             }
         }
 
-        override fun bind(item: Safe, payloads: List<Any>?) {
-            itemView.layout_safe_item_address.text = item.address.asEthereumAddressString()
-            itemView.layout_safe_item_name.text = item.name
-            itemView.layout_safe_item_name.visibility = if (item.name.isNullOrEmpty()) View.GONE else View.VISIBLE
+        private var currentEntry: Safe? = null
+
+        override fun castedBind(data: Safe, payloads: List<Any>?) {
+            currentEntry = data
+            itemView.layout_safe_item_address.text = data.address.asEthereumAddressString()
+            itemView.layout_safe_item_name.text = data.name
+            itemView.layout_safe_item_name.visibility = if (data.name.isNullOrEmpty()) View.GONE else View.VISIBLE
+        }
+
+        override fun unbind() {
+            currentEntry = null
+            super.unbind()
         }
 
         override fun onClick(v: View?) {
-            safeSelection.onNext(items[adapterPosition])
+            currentEntry?.let { safeSelection.onNext(it) }
+        }
+    }
+
+    inner class PendingViewHolder(itemView: View) : CastingViewHolder<PendingSafe>(PendingSafe::class.java, itemView) {
+
+        private val disposables = CompositeDisposable()
+
+        private var currentEntry: PendingSafe? = null
+
+        override fun castedBind(data: PendingSafe, payloads: List<Any>?) {
+            currentEntry = data
+            itemView.layout_pending_safe_item_name.text = data.name
+        }
+
+        @OnLifecycleEvent(Lifecycle.Event.ON_START)
+        fun start() {
+            // Make sure no disposable are left over
+            disposables.clear()
+            val pendingSafe = currentEntry ?: return
+            disposables += safeRepository.observeDeployStatus(pendingSafe.hash.asTransactionHash())
+                    .observeOn(AndroidSchedulers.mainThread())
+                    .subscribe(this::safeDeployed, Timber::e)
+
+        }
+
+        private fun safeDeployed(safeAddress: String) {
+            Timber.i("#######", safeAddress)
```

Comment:
Missing string format here?

Reply:
actually I should update the ui ... or just to nothing either way